From fcf507bf41eb73489ad3eeaa06958281e63182bd Mon Sep 17 00:00:00 2001
From: Jianhua Xie <jianhua.xie@freescale.com>
Date: Tue, 22 Apr 2014 09:17:49 +0800
Subject: [PATCH 0933/1089] bonding: make LAG codes to fit different kernel

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

Linux kernel API bond_for_each_slave of bonding.h has different params
along with different kernel version as below table:

Older version:		bond_for_each_slave(bond, pos, int cnt),
3.11.0-rc1(dec1e90e8):	bond_for_each_slave(bond, pos),
3.12.0-rc1(9caff1e7b):	bond_for_each_slave(bond, slave, list_head *iter)

This patch is making LAG codes to fit different kernel version.

Signed-off-by: Jianhua Xie <jianhua.xie@freescale.com>
Change-Id: Ifd365cd232aba67a437b156568d88c6e16c44c1a
Reviewed-on: http://git.am.freescale.net:8181/11293
Reviewed-by: Dongsheng Wang <dongsheng.wang@freescale.com>
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Jiafei Pan <Jiafei.Pan@freescale.com>
Reviewed-by: Scott Wood <scottwood@freescale.com>
---
 drivers/net/bonding/hw_distribution.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/bonding/hw_distribution.c b/drivers/net/bonding/hw_distribution.c
index 1403659..e87d016 100644
--- a/drivers/net/bonding/hw_distribution.c
+++ b/drivers/net/bonding/hw_distribution.c
@@ -348,10 +348,9 @@ static bool is_dpa_eth_port(struct net_device *netdev)
 bool are_all_slaves_linkup(struct bonding *bond)
 {
 	struct slave *slave;
-	struct list_head *iter;
 
 	read_lock(&bond->lock);
-	bond_for_each_slave(bond, slave, iter)
+	bond_for_each_slave(bond, slave)
 		if (!(SLAVE_IS_OK(slave))) {
 			read_unlock(&bond->lock);
 			return false;
-- 
2.0.2

